pinot-plugins/pinot-input-format/pinot-avro-base/src/main/java/org/apache/pinot/plugin/inputformat/avro/AvroUtils.java

```diff
@@ -134,18 +136,19 @@ public static Schema getPinotSchemaFromAvroDataFile(File avroDataFile)
    * @param complexType if allows complex-type handling
    * @param unnestFields the fields to unnest
    * @param delimiter the delimiter separating components in nested structure
+   * @param collectionToJsonMode to mode of converting collection to JSON string
    * @return Pinot schema
    */
   public static Schema getPinotSchemaFromAvroSchemaFile(File avroSchemaFile,
       @Nullable Map<String, FieldSpec.FieldType> fieldTypeMap, @Nullable TimeUnit timeUnit, boolean complexType,
-      List<String> unnestFields, String delimiter)
+      List<String> unnestFields, String delimiter, ComplexTypeConfig.CollectionToJsonMode collectionToJsonMode)
```

Comment:
(nit)
- Can we pass all three unnestFields, delimiter and collectionToJsonMode as one config that encapsulates all the complex type config since I believe unnestFields and delimiter are also extracted from what user configured

- Suggest renaming collectionToJson -> convertCollectionNotUnnestedToJson or something similar which is slightly more descriptive

Reply:
# 1. But these are parsed from command-line configs? 
# 2 I thought about `convertCollectionNotUnnestedToJson`, but feel it's a bit verbose :)

Reply:
For (1), you might want to change the name to `fieldsToUnnest`. I see the admin command has unnestFields as well. I missed it earlier. 
For (2), I prefer the verbose name. `collectionToJsonMode` doesn't really imply that this is for collections not unnested

Reply:
okay. `fieldsToUnnest ` sounds a better name to me. I'll address it in another PR, coz it's used across the board, including in the complexTypeConfig.

I'll go with `collectionNotUnnestedToJson`. Similarly,  I'll do the rename in the same PR as above.